core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -215,22 +229,61 @@ void dispatchRequest(final RestRequest request, final RestChannel channel, final
     }
 
     /**
+     * If a request contains content, this method will return {@code true} if the {@code Content-Type} header is present, matches an
+     * {@link XContentType} or the request is plain text, and content type is required. If content type is not required then this method
+     * returns true unless a content type could not be inferred from the body and the rest handler does not support plain text
+     */
+    boolean checkForContentType(final RestRequest restRequest, final RestHandler restHandler) {
+        if (restRequest.getXContentType() == null) {
+            if (restHandler != null && restHandler.supportsPlainText()) {
```

Comment:
I assumed that we failed earlier in the handler is null. It makes sense to do so but it if it becomes a bigger change  we can make it as a followup